mir-utils/b2ctab.c

```diff
@@ -2,6 +2,12 @@
 
 #include "mir.h"
 
+#ifdef _WIN32
+/* <stdio.h> provides _fileno */
+#include <fcntl.h> /* provides _O_BINARY */
+#include <io.h>    /* provides _setmode */
+#endif
```

Comment:
```c
#define set_filemode_binary(F) _setmode (_fileno (F), _O_BINARY)
#else
#define set_filemode_binary(F) 0
```
you could add this here and then use set_filemode_binary(stdin) below without more ugly ifdefs.
i personally find the technique more appealing.